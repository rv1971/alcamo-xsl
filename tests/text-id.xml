<?xml version="1.0" encoding="UTF-8"?>

<?xml-stylesheet href="#xslt" type="text/xml"?>

<a:doc xmlns:a="tag:rv1971@web.de,2021:alcamo-xsl#">
  <xsl:stylesheet
      xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
      xml:id="xslt"
      version="1.0">
    <xsl:import href="../xsl/text.xsl"/>

    <xsl:template match="/">
      <a:result>
        <xsl:apply-templates select="/a:doc/a:data"/>

        <xsl:apply-templates select="/a:doc/a:data/a:item/@*"/>
      </a:result>
    </xsl:template>

    <xsl:template match="a:item">
      <a:item>
        <xsl:value-of select="."/>
        <xsl:value-of select="@*"/>
        <xsl:value-of select="' '"/>
        <xsl:apply-templates select="." mode="a:id"/>
      </a:item>
    </xsl:template>

    <xsl:template match="@id|@xml:id">
      <a:item>
        <xsl:value-of select="."/>
        <xsl:value-of select="' '"/>
        <xsl:apply-templates select="." mode="a:id"/>
      </a:item>
    </xsl:template>
  </xsl:stylesheet>

  <a:data>
    <a:item id="foo"/>
    <a:item xml:id="bar"/>
    <a:item xml:id="baz" id="qux"/>
    <a:item>BAZ: Qux!</a:item>
    <a:item>FOO_BAR! baz"#$%&amp;'()*+,./:;&lt;&gt;?@[\]^`{|}~</a:item>
  </a:data>
</a:doc>
