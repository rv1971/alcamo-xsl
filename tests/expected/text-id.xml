<?xml version="1.0"?>
<a:result xmlns:a="tag:rv1971@web.de,2021:alcamo-xsl#">
  <a:item>foo data.foo</a:item>
  <a:item>bar bar</a:item>
  <a:item>baz baz</a:item>
  <a:item>BAZ: Qux! baz-qux</a:item>
  <a:item>FOO_BAR! baz"#$%&amp;'()*+,./:;&lt;&gt;?@[\]^`{|}~ foo_bar-baz.</a:item>
  <a:item>foo data.foo</a:item>
  <a:item>bar bar</a:item>
  <a:item>baz baz</a:item>
  <a:item>qux data.qux</a:item>
</a:result>
